temporal_sqlalchemy/bases.py

```diff
@@ -21,9 +23,9 @@
 
 
 class EntityClock(object):
-    tick = sa.Column(sa.Integer, primary_key=True, autoincrement=False)
-    timestamp = sa.Column(sa.DateTime(True),
-                          server_default=sa.func.current_timestamp())
+    id = sa.Column(sap.UUID(as_uuid=True), default=uuid.uuid4, primary_key=True)
+    tick = sa.Column(sa.Integer, nullable=False)
+    timestamp = sa.Column(sa.DateTime(True), server_default=sa.func.current_timestamp())
```

Comment:
It's not up to SQLAlchemy.  It's an implicit requirement in Postgres.

See the Postgres's documentation for [CREATE TABLE](https://www.postgresql.org/docs/9.5/static/sql-createtable.html):

"The PRIMARY KEY constraint specifies that a column or columns of a table can contain only unique (non-duplicate), nonnull values."